Python 3.12 incompatibility: "ModuleNotFoundError: No module named 'imp'" while building subunit dependency using Conan

`hyperonc` module is affected by https://bugs.launchpad.net/subunit/+bug/2052948

When trying to build it in environment with Python 3.12 the following error happens:
```
...
libtool: install: chmod 644 ~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378//lib/libcppunit_subunit.a
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'imp'
libtool: install: ranlib ~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378//lib/libcppunit_subunit.a
make[1]: *** [Makefile:957: install-pkgpythonPYTHON] Error 1
make[1]: *** Waiting for unfinished jobs....
libtool: warning: remember to run 'libtool --finish //lib'
make[1]: Leaving directory '~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/src'
make: *** [Makefile:1447: install-am] Error 2
ERROR: subunit/1.4.0: Error in package() method, line 147
 autotools.install()
 ConanException: Error 2 while executing make install 'INSTALLARCHLIB=~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378/lib' 'INSTALLSITEARCH=~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/build-release/archlib' 'INSTALLVENDORARCH=~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/build-release/archlib' 'INSTALLSITEBIN=~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378/bin' 'INSTALLSITESCRIPT=~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378/bin' 'INSTALLSITEMAN1DIR=~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/build-release/share/man/man1' 'INSTALLSITEMAN3DIR=~/.conan/data/subunit/1.4.0/_/_/build/095274926d7784555b5378b8638d408c0c4d0378/build-release/share/man/man3' DESTDIR=~/.conan/data/subunit/1.4.0/_/_/package/095274926d7784555b5378b8638d408c0c4d0378 -j16
```

--- a/README.md
+++ b/README.md
@@ -83,7 +83,7 @@ page](https://www.rust-lang.org/tools/install). Make sure your
 Rust (see the Notes at the installation page).
 
   Requirements for building C and Python API
-  * Python3 and Python3-dev (3.7 or later, but not 3.12)
+  * Python3 and Python3-dev (3.7 or later)
   * Pip (23.1.2 or later)
   * GCC (7.5 or later)
   * CMake (3.19 or later)

--- a/c/conanfile.txt
+++ b/c/conanfile.txt
@@ -3,3 +3,6 @@ libcheck/0.15.2
 
 [generators]
 cmake
+
+[options]
+libcheck:with_subunit=False

--- a/python/pyproject.toml
+++ b/python/pyproject.toml
@@ -6,7 +6,7 @@ build-backend = "setuptools.build_meta"
 name = "hyperon"
 description = "Hyperon API in Python"
 readme = "README.md"
-requires-python = ">=3.7,<3.12"
+requires-python = ">=3.7"
 keywords = ["metta", "hyperon", "opencog"]
 license = {text = "MIT License"}
 classifiers = [

